### PR TITLE
Label file support

### PR DESCRIPTION
**Pull request by aanm**
Implemented label-file support for docker-compose.yml

I think that everything is working, unless some tests fail.
@aanand what do you think?

### CODE REVIEW

**Comment by aanm** [comment]
The `extends` option deprecate this PR

**Comment by aanand** [comment]
I wouldn't say it deprecates it - `labels_file` would still be useful for setting the same labels across multiple heterogeneous services in a file - but it's true that it adds slightly less value now.

**Comment by thaJeztah** [comment]
Also, labels-file is something that docker itself supports so might still be desirable to bring parity with docker.

**Comment by aanand** [comment]
Yep - if people are using `docker run --labels-file` and want to switch to Compose, they're gonna want a replacement for it.

**Comment by aanm** [comment]
Right now the option is not available on CLI, do you think it should be?

**Comment by aanand** [comment]
Thanks for working on this. No, it's not imperative that we support it in the CLI.

The labels file support as you've implemented it differs significantly from Docker's `--label-file` support. I see the value in it, but it's more than we need for feature parity, and as far as I can tell the file format isn't backwards-compatible with the Docker CLI. It'd be better for this PR to just support Docker's simple label file format.

**Comment by aanm** [comment]
@aanand I see that `env_file` doesn't seem to have the same behavior of docker's `env-file`. It seems like `compose` deals with all env by himself without giving `docker daemon` a chance of know that `env-file` was used on `compose`. Am I wrong?

**Comment by thaJeztah** [comment]
@aanm the point is, that users should be able to create an env-file, use it with docker, and when creating their "composition", use *the same file* in docker-compose. For that to work, docker-compose and docker should use the same file-format.

**Comment by aanm** [comment]
@thaJeztah ah! I was seeing things wrongly. I'm going to make necessary changes then.

**Comment by aanm** [comment]
@aanand Will you merge this in the next days? Please tell me so I will solve the conflicts.

File: compose/config.py [hunk 1]
```diff
@@ -344,6 +346,63 @@ def env_vars_from_file(filename):
...
+    if not os.path.exists(filename):
+        raise ConfigurationError("Couldn't find label file: %s" % filename)
+
+    labels = {}
+    for line in open(filename, 'r'):
+        line = line.strip()
+        if line and not line.startswith('#'):
+            k, v = split_label(line)
+            labels[k] = v
+    return labels
```
Comment (dnephin) [inline compose/config.py]: A lot of this code is almost identical to  the `env_file` equivalent. We should just add a parameter to those functions and re-use the same code. I think that will make it easier to rebase.

**Comment by aanand** [comment]
Sorry for the delay - if you rebase this and make the refactoring @dnephin, we'll merge it.

**Comment by aanm** [comment]
@aanand
I've changed the code to be updated with the current code.
The tests errors are caused by this pull request?

**Comment by aanand** [comment]
Yep, looks like you've got some formatting issues:

```
compose/config/config.py:342:1: E302 expected 2 blank lines, found 1
compose/config/config.py:346:11: E225 missing whitespace around operator
compose/config/config.py:350:5: F821 undefined name 'env'
compose/config/config.py:351:64: F821 undefined name 'env'
compose/config/config.py:353:1: E302 expected 2 blank lines, found 1
compose/config/config.py:543:1: E302 expected 2 blank lines, found 1
compose/config/config.py:677:1: F811 redefinition of unused 'split_label' from line 543
tests/integration/service_test.py:862:5: E303 too many blank lines (2)
```

**Pull request closed without merging**